core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -376,19 +407,16 @@ private void applyConfiguration(CreateContainerCmd createCommand) {
             String alias = linkEntries.getKey();
             LinkableContainer linkableContainer = linkEntries.getValue();
 
-            Set<Link> links = dockerClient.listContainersCmd().exec().stream()
-                    .filter(container -> container.getNames()[0].endsWith(linkableContainer.getContainerName()))
-                    .map(container -> new Link(container.getNames()[0], alias))
+            Set<Link> links = dockerClient.listContainersCmd()
+                    .withStatusFilter("running")
```

Comment:
👍

Reply:
Ah yes - forgot to mention this in the summary. It seems that we had a redundant step here. While we're planning to kill off links, it seemed like a good fix to make all the same.